### ISSUE TITLE
Setup DNS and redirect for x-k8s.io

### ISSUE THREAD

**Issue by dims** [kubernetes/k8s.io#320]
To support the new efforts around API/CRDs https://github.com/kubernetes/community/pull/3875

**Comment by dims** [comment]
```
dims 1:36 PM
@thockin another one i missed mentioning this morning. CNCF now has the x-k8s.io domain we requested. what do we do next with it? Could we setup a redirect or something to a page that we own?
thockin 1:37 PM
uggh, are we actually proceeding with that proposal
do we want it to resolve to somethng?  To what?  We can own the domain for purposes of CRD scoping without actually DOING anything with it
stp 1:38 PM
Damn wanted to use it as testing ground for redirector. Side benefit it would by default redirect to kubernetes.io ^^
jdetiber:vmware: 1:39 PM
Netlify redirects for api documentation for CRDs defined under a *.x-k8s.io api group? (edited) 
:+1:
1

dims 1:41 PM
@thockin we could just point that to k/enhancements repo for now
munnerz 1:43 PM
Netlify redirects for api documentation for CRDs defined under a *.x-k8s.io api group? (edited)
I know this of interest for kind - we currently have kind.sigs.k8s.io/v1alpha1 redirect to our API docs, and I know the intent is to have kind.x-k8s.io/v1alpha1 redirect too… :see_no_evil: (cc @bentheelder)
stp 1:43 PM
This would be something I could do with the redirector. Could also be something that's non breaking, while we test redirect ideas?
Could bounce of needs and ideas with munnerz, if that makes sense.
dims 1:44 PM
works for me @stp
@stp whoa! multiple stp`s :slightly_smiling_face:
jdetiber:vmware: 1:44 PM
It would be nice from my perspective for cluster-api to be able to have API docs published under the api group URL and user docs under *.sigs.k8s.io
stp 1:45 PM
Haha multiple of us. Wouldn't have guesses.
dims 1:45 PM
@jdetiber agree, but that’s a longer discussion. i wanted something quick to take ownership of that domain
:+1:
1

stp 1:46 PM
OK so proposal: Use TXTDirect DNS records to redirect the cluster-api to the api docs under the new domain with the hosted service to test out the DNS side of things and do the self-hosting and actual go.k8s.io in parallel. Aka test with new non breaking stuff and do the rest on the side/next step.
bentheelder:keanu: 1:46 PM
[catching up]
yeah, that proposal went forward AFAIK.
kind will do whatever is recommended. I'm with @jdetiber's thinking re: docs etc. for now, but also +1 on doing something quicker for now to get ownership up :sweat_smile: (edited) 
thockin 1:47 PM
it's NOT true for k8s.io groups that the GVK leads to docs
munnerz 1:48 PM
yep, true ^ but it’s definitely a nice feature, and it’s something that kind has managed :slightly_smiling_face:
:+1:
1

dims 1:48 PM
“several people are typing” :slightly_smiling_face: LOL
:sweat_smile:
2
:joy:
1

bentheelder:keanu: 1:48 PM
yeah, definitely not necessary, just a nice to have :slightly_smiling_face:
thockin 1:48 PM
storage.k8s.io/StorageClass for example, becasue we did not carve out a sub-space
munnerz 1:48 PM
whether we should have achieved it, I guess, is another topic
new messages
bentheelder:keanu: 1:49 PM
we could do that for storageclass etc. I did float the idea with sig docs but it doesn't work for like apps/v1
thockin 1:49 PM
well, that evolution all happened ad hoc, with a bit too little planning, so , yeah probably
It's a bg step in PITA factor to go from "nothing" to "DNS resolves and "somethign".
the DNS zone set5up is not scruipted yet
have to run off  right now, someone take ownership and write somethng
dims 1:54 PM
@stp @munnerz - you have the pen! :slightly_smiling_face:

```

**Comment by stp-ip** [comment]
After a bit of discussion various options to go forward were mentioned:
* full automation from zone creation to separate octo-dns deploy
* manual zone creation + current octo-dns deploy
* discuss usage of x-k8s.io and propose usage without any implementation work

As it currently stands with manual time taking away already slim time from key people and full recreation being partial dupliacte implementation work, the direction currently is to write down a proposal for the usage of x-k8s.io.

@munnerz ideas?

**Comment by stp-ip** [comment]
Zones are now managed via our DNS as done in #526
Missing is adding useful redirects. I would defer this once we have k8s.io migrated and might even push this after we have the redirector running.

**Comment by spiffxp** [comment]
/close
Completed via:
- https://github.com/kubernetes/k8s.io/pull/2427
- https://github.com/kubernetes/k8s.io/pull/2437
